server/app/controllers/admin/NorthStarQuestionController.java

```diff
@@ -0,0 +1,69 @@
+package controllers.admin;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.CiviFormProfile;
+import auth.ProfileUtils;
+import com.google.common.collect.ImmutableList;
+import controllers.CiviFormController;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.pac4j.play.java.Secure;
+import play.i18n.Lang;
+import play.i18n.Messages;
+import play.i18n.MessagesApi;
+import play.mvc.Http;
+import play.mvc.Http.Request;
+import play.mvc.Result;
+import repository.VersionRepository;
+import services.applicant.ApplicantPersonalInfo;
+import services.applicant.ApplicantPersonalInfo.Representation;
+import services.question.exceptions.InvalidQuestionTypeException;
+import services.question.types.QuestionType;
+import views.admin.questions.NorthStarQuestionPreview;
+
+/** Controller for rendering inputs for questions. */
+public final class NorthStarQuestionController extends CiviFormController {
+  private final NorthStarQuestionPreview northStarQuestionPreview;
+  private final Messages messages;
+
+  @Inject
+  public NorthStarQuestionController(
+      ProfileUtils profileUtils,
+      VersionRepository versionRepository,
+      NorthStarQuestionPreview northStarQuestionPreview,
+      MessagesApi messagesApi) {
+    super(profileUtils, versionRepository);
+    this.northStarQuestionPreview = checkNotNull(northStarQuestionPreview);
+    this.messages = messagesApi.preferred(ImmutableList.of(Lang.defaultLang()));
+  }
+
+  @Secure
+  public Result sampleQuestion(Request request, String questionType) {
+    Representation representation = Representation.builder().build();
+    ApplicantPersonalInfo api = ApplicantPersonalInfo.ofGuestUser(representation);
+    Optional<CiviFormProfile> profile = profileUtils.currentUserProfile(request);
+
+    QuestionType questionTypeEnum;
+    try {
+      questionTypeEnum = QuestionType.fromLabel(questionType);
+    } catch (InvalidQuestionTypeException e) {
+      System.out.println("Invalid question type: " + questionType);
+      return badRequest("Invalid question type: " + questionType);
+    }
+
+    NorthStarQuestionPreview.Params params =
+        NorthStarQuestionPreview.Params.builder()
+            .setRequest(request)
+            .setApplicantId(0l)
+            .setApplicantPersonalInfo(api)
+            .setProfile(profile.orElse(null))
+            .setType(questionTypeEnum)
+            .setMessages(messages)
+            .build();
+
+    String content = northStarQuestionPreview.render(params);
```

Comment:
this Controller should probably check the flag value. I know the htmx is shown only if the flag is enabled, but a user could hit the route directly

Reply:
What does the user expect to see if they manually input a URL like http://foo.gov/samplequestion/text ?

Here's http://localhost:9000/samplequestion/text

<img width="799" alt="Screenshot 2024-06-27 at 9 08 34 AM" src="https://github.com/civiform/civiform/assets/23390490/f3cb8670-d2e3-41ed-9f51-5e95d3757428">

Do we really want to spend time adding a non-NorthStar codepath in NorthStarQuestionController.java and testing it, just for this edge case? (I'd rather not)

Reply:
I'd expect a 404 with the flag disabled. I know it's not the most important in a world in which it's open source, but it's just a norm to not allow users to see any feature flagged content without the flag enabled. I'll defer to the rest of the team on it though. can you ask in eng-general?